src/conf.c

```diff
@@ -66,9 +66,10 @@
 #define PASSWORD "([^@]*)"
 #define IP "((([0-9]{1,3})\\.){3}[0-9]{1,3})"
 #define IPMASK "(" IP "(/" DIGIT "+)?)"
+#define IPV6_SCOPE "((%[^ \\t\\/]{1,16})?)"
 #define IPV6 "(" \
-        "(([0-9a-f:]{2,39}))|" \
-        "(([0-9a-f:]{0,29}:" IP "))" \
+        "([0-9a-f:]{2,39})" IPV6_SCOPE "|" \
```

Comment:
won't removal of the outer parens pair break other uses of the regex ?

Reply:
The definition von IPV6 is used for listen, allow, deny, bind and upstream. handle_listen(), handle_allow(), handle_deny() and handle_bind() all use match[2], so the additional capture groups do not break these handlers.

Only handle_upstream() had to be changed, because the additional capture grups requred skipping more captures to get the port number.

I wish, standard regex would support non-capturing groups like perl.